### ISSUE TITLE
Enhancement request: speed limit per connection (or a variant thereof)

### ISSUE THREAD

**Issue by Morganlej** [owncloud/client#4285]
Scenario: Client 2.x with synchronisations to local and internet servers.

People like us who have a slow internet line do not want it to be swamped by synchronisations, so i want to set the speed limit low.

But we also have a local owncloud server on LAN serving large files over gigabit connections, and we do not want to limit the speed of that!  Also, the local connections should update quickly and not wait "ages" until slow transfers to external servers are done...

I think there should be divided like one limit set for local servers on LAN, and one for internet.
Also they should operate independently, like if i set internet upload max to 30 kB/s, it should keep running that, even while it is uploading to local server. (if/when concurrency is supported)

Maybe, to cater for odd configurations and needs - and maybe it is more clear - have a separate speed limit set per server.  But for people who connect to several internet servers a total limit of all connections to servers on internet would be nice.

(As a side note, it would be nice if some prioritising could be done, but it works as per now i manually pause what is less important at the moment, depending on what project i work upon.)

**Comment by danimo** [comment]
@jancborchardt

**Comment by jancborchardt** [comment]
Shouldn’t local servers always be connected to with the fastest way? Or is there any reason why you wouldn’t want to do that?

**Comment by Morganlej** [comment]
Always good to have options even we can not think of reason right now...
Maybe there is some bottleneck or redirect, or...

**Comment by jancborchardt** [comment]
> Always good to have options even we can not think of reason right now

@Morganlej sorry, but that’s not the way we operate. If we would, ownCloud would be full of features and options no one really needs, and eventually no one would use it because of all the clutter. If there is no clear use case (or multiple ones), there is no reason to implement and maintain it.

**Comment by Morganlej** [comment]
Yes full speed on LAN far better than always limit (provided there is an always working method to tell if a server is on LAN or not)
The other part to make LAN sync work while a hour long internet sync is running is of course that they should be able to operate concurrently. I dont know the plans for that.
